inspire_schemas/utils.py

```
@@ -278,3 +282,90 @@ def _is_roman_numeral(suffix):
         if part)
 
     return final_name
+
+
+def format_date(year, month=None, day=None):
+    """Format a (potentially incomplete) date given its numeric components.
+
+    If a component is a numeric `str` or an `int` it will be used, otherwise it
+    will be ignored.
+
+    If parsing of textual data is needed, :func:`normalize_date` should be used
+    instead.
+
+    Returns:
+        str: a formatted date, in the form YYYY-MM-DD, YYYY-MM or YYYY
+            (depending on the information present in the date).
+
+    Raises:
+        ValueError: when year does not contain valid data.
```

Not sure about that, I think it is more pythonic to raise an exception instead of returning an empty string.